platform/windows/vulkan_context_win.cpp

```diff
@@ -55,7 +55,8 @@ Error VulkanContextWindows::window_create(DisplayServer::WindowID p_window_id, D
 	return _window_create(p_window_id, p_vsync_mode, surface, p_width, p_height);
 }
 
-VulkanContextWindows::VulkanContextWindows() {
+VulkanContextWindows::VulkanContextWindows(const uint8_t p_frame_count, const uint8_t p_swapchain_count) :
```

Comment:
```suggestion
VulkanContextWindows::VulkanContextWindows(uint8_t p_frame_count, uint8_t p_swapchain_count) :
```

Reply:
Is there a reason to remove the constness from function declaration?

I see Godot barely uses const-correctness and it increases the mental load to parse the code by 1000x. I often have to add const to variables while analyzing to help understanding it, and then remove it.

Reply:
* The declaration is non-const: `VulkanContextWindows(uint8_t p_frame_count, uint8_t p_swapchain_count);`
* Primitive types don't need constnest, so it isn't relevant, and only make the code more busy

Reply:
> The declaration is non-const: VulkanContextWindows(uint8_t p_frame_count, uint8_t p_swapchain_count);, at minimum these should match

The reason for that is not to tie the API to a specific detail that may change in the future.

> Primitive types don't need constnest, so it isn't relevant

Yes, it's very relevant. Knowing that `p_swapchain_count` can never change from that line onward is a heavy alleviation on the mental load to parse & understand what a snippet of code is doing.

Reply:
> Yes, it's very relevant. Knowing that p_swapchain_count can never change from that line onward is a heavy alleviation on the mental load to parse & understand what a snippet of code is doing.

Generally I agree

Reply:
> What do you mean? Why shouldn't the header match the source?

The following:

```cpp
// Header
void my_function( int arg );

// Source
void my_function( const int arg )
{
}
```

Indicates:

1. The API/ABI does not impose restrictions.
2. The source uses const to indicate the developers the value won't be changing.
   - If later the that needs to be lifted for some reason, the devs can do it.


The following:

```cpp
// Header
void my_function( const int arg );

// Source
void my_function( const int arg )
{
}
```

1. Now you can't de-const without potentially breaking the API/ABI. You need to use deprecation, e.g. add my_function2.


And finally the following:

```cpp
// Header
void my_function( int arg );

// Source
void my_function( int arg )
{
}
```

1. The API/ABI has no restrictions.
2. Parsing the source is more mentally taxing.

Reply:
Sure, it is legal c++, but I don't think we *ever* do this, and it is confusing to have different declaration and definition

Reply:
We use a `p_` prefix for all parameters (and `r_` for out-values), so if you assign a `p_` parameter, it's usually obvious from reading the source (and you probably won't ever be doing this anyway).

Reply:
I think that is kind of besides the point. ``p_`` means " This is an input parameter _and_ I will not be using this to return a variable. It does not mean "this variable will not be modified in the scope of this function". Adding a const ensures that the parameter won't be changed inside the function. For basic types it may not improve performance as it does with complex types, but that doesn't mean that it isn't beneficial. Readability is often more important than performance for us anyway.

Reply:
I think the more important point is:
Do we allow the declaration and definition to differ on the specifics of arguments

I do not believe we have any cases of this anywhere, but I might be wrong

Reply:
I'd favor consistency here. The `const` in by-value parameters (primitive type or not) is surprising enough to hamper readability while not letting the reader know anything the `p_` prefix already tells.

I've changed my mind a bit over time on the usage of `const` for locals. If it helps following the code, I'm no longer against it even if I see downsides. However, in cases like this I can't see the benefit.